src/include/daos/placement.h

```diff
@@ -136,7 +136,8 @@ int pl_obj_find_reint(struct pl_map *map,
 
 typedef struct pl_obj_shard *(*pl_get_shard_t)(void *data, int idx);
 
-int pl_select_leader(daos_unit_oid_t *oid, int nr, bool for_tgt_id,
-		     pl_get_shard_t pl_get_shard, void *data, uint32_t *leader);
+int pl_select_leader(daos_obj_id_t oid, uint32_t shard_idx, int shards_nr,
+		     bool for_tgt_id, pl_get_shard_t pl_get_shard, void *data,
+		     uint32_t *leader);
```

Comment:
why don't we just return pl_obj_shard? It could make the parameter cleaner

Reply:
I will change it.